src/test_application/flight.cpp

```diff
@@ -0,0 +1,167 @@
+
+    // Create the application
+    magnumApp = std::make_unique<OSPMagnum>(args, ospApp);
+
+    // Configure the controls
+    config_controls(*magnumApp);
+
+    // Create an ActiveScene
+    osp::active::ActiveScene& scene = magnumApp->scene_add("Area 1");
+
+    // Register dynamic systems for that scene
+
+    auto &sysPhysics = scene.dynamic_system_add<osp::active::SysPhysics>(
+                "Physics");
+    auto &sysWire = scene.dynamic_system_add<osp::active::SysWire>(
+                "Wire");
+    auto &sysDebugRender = scene.dynamic_system_add<osp::active::SysDebugRender>(
+                "DebugRender");
+    auto &sysArea = scene.dynamic_system_add<osp::active::SysAreaAssociate>(
+                "AreaAssociate", uni);
+    auto &sysVehicle = scene.dynamic_system_add<osp::active::SysVehicle>(
+                "Vehicle");
+    auto &sysPlanet = scene.dynamic_system_add<planeta::active::SysPlanetA>(
+                "Planet");
+    auto &sysGravity = scene.dynamic_system_add<osp::active::SysFFGravity>(
+                "FFGravity");
+
+    // Register machines for that scene
+    scene.system_machine_add<SysMachineUserControl>("UserControl",
+            magnumApp->get_input_handler());
+    scene.system_machine_add<SysMachineRocket>("Rocket");
+
+    // Make active areas load vehicles and planets
+    sysArea.activator_add(satAA, sysVehicle);
+    sysArea.activator_add(satPlanet, sysPlanet);
+
+    // create a Satellite with an ActiveArea
+    Satellite areaSat = uni.sat_create();
+
+    // assign sat as an ActiveArea
+    UCompActiveArea &area = satAA->add_get_ucomp(areaSat);
+
+    // Link ActiveArea to scene using the AreaAssociate
+    sysArea.connect(areaSat);
+
+    // Add a camera to the scene
+
+    // Create the camera entity
+    ActiveEnt camera = scene.hier_create_child(scene.hier_get_root(),
+                                                       "Camera");
+    auto &cameraTransform = scene.reg_emplace<ACompTransform>(camera);
+    auto &cameraComp = scene.get_registry().emplace<ACompCamera>(camera);
+
+    cameraTransform.m_transform = Matrix4::translation(Vector3(0, 0, 25));
+    scene.reg_emplace<ACompFloatingOrigin>(camera);
+
+    cameraComp.m_viewport
+            = Vector2(Magnum::GL::defaultFramebuffer.viewport().size());
+    cameraComp.m_far = 4096.0f;
+    cameraComp.m_near = 0.125f;
+    cameraComp.m_fov = 45.0_degf;
+
+    cameraComp.calculate_projection();
+
+    // Add the debug camera controller to the scene. This adds controls
+    auto camObj = std::make_unique<DebugCameraController>(scene, camera);
+
+    // Add a ACompDebugObject to camera to manage camObj's lifetime
+    scene.reg_emplace<ACompDebugObject>(camera, std::move(camObj));
+
+    // Starts the game loop. This function is blocking, and will only return
+    // when the window is  closed. See OSPMagnum::drawEvent
+    magnumApp->exec();
+
+    // Close button has been pressed
+
+    std::cout << "Magnum Application closed\n";
```

Comment:
Probably want something like boostlog or other logging framework. Using std::court is yucky

Reply:
Adding boost log should probably be in a separate PR. I haven't got to doing it yet, as it's a task that I'm thinking a contributor can get easily started on.